### DISCUSSION TITLE
Python system wide install

### DISCUSSION THREAD

**Discussion by Josh65-2201** [microsoft/winget-pkgs#24389] [Q&A] [answered]
Currently there doesn't seem to be a way of installing python 3 in a system wide environment. How if its already a thing can I do it.

**Comment by OfficialEsco** [comment, marked as answer]
There is currently not a switch for it in the manifest, if someone gets to it before me here is the InstallerSwitches https://docs.python.org/3/using/windows.html

You can do `--override "/quiet PrependPath=1 InstallAllUsers=1`, later you can do `winget install python3 --scope machine`

For the one adding the metadata, we need `Custom: InstallAllUsers=1` and `InstallLocation: TargetDir=<INSTALLPATH>`

**Reply by OfficialEsco** [reply]
It appears the Python installer is Burn which means some of the parameters can be automated by winget-cli (Hopefully more in the future).
I cannot confirm that the `--location` switch works because it does not seem to work with `--manifest` however the `--scope` switch works as expected.

If you prefer Machine Wide installers you can add it to `winget settings`, it will then install it machine wide if there is a installer for it in the manifest.

PR; https://github.com/microsoft/winget-pkgs/pull/24437
